### PR TITLE
redirect home on 404

### PR DESCRIPTION
**Pull request by perryr16**
#### Any background context you want to provide?
When an unknown URL is entered a 404 error is initiated. SEED loads the sidebar, header, and 404 content. With the 404 page loaded the sidebar fails to lookup an active org and prompts the user to create a new one (similar to a user's first time visiting seed). Upon filling out a new organization, the user will be redirected to the home page where they will have a new organization attached to their profile.

#### What's this PR do?
Instead of sending users to a 404 or 500 page, all 404 and 500 errors will redirect to the home page '/' with an alert message in the header area indicating the error and reason.

The backend will redirect to '/' with a query param `?http_error=404` or `500`. The query parameter is visible in the url bar initially but soon disappears as the home page continues to load.

#### How should this be manually tested?
To test locally in docker a few settings need to be modified. 
- /config/settings/docker_dev.py --> change `DEBUG` to `False` 
- /docker-compose.dev.yml --> change `web`...`command:` to  `python3 manage.py runserver --insecure --settings=config.settings.docker_dev 0.0.0.0:80`

Go to any page that does not exist. You should be redirected to "/app/"

#### What are the relevant tickets?
#3140 

#### Screenshots (if appropriate)

https://user-images.githubusercontent.com/58446472/156236768-d06ced06-7dce-4934-95c3-f84ec4ee55b1.mov

### CODE REVIEW

**Comment by perryr16** [comment]
Future Options
- Check the url before opening the `create_organization_modal`. If the `window.location.path !== '/app/'` then do not prompt the user for a new organization.
- Register the 404 or 500 has been triggered inside the `menu_controller` and prevent the `creaet_organization_modal` from opening

File: seed/static/seed/js/controllers/menu_controller.js [hunk 1]
```diff
@@ -280,6 +280,15 @@ angular.module('BE.seed.controller.menu', [])
           });
         }
       };
+
+      if ($location.search().http_error == 404) {
```
Comment (perryr16) [inline seed/static/seed/js/controllers/menu_controller.js:284]: retreive query parameter

File: seed/views/main.py [hunk 2]
```diff
@@ -99,29 +99,23 @@ def version(request):
...
 
 def error404(request, exception):
-    # Okay, this is a bit of a hack. Needed to move on.
     if '/api/' in request.path:
         return JsonResponse({
             "status": "error",
             "message": "Endpoint could not be found",
         }, status=status.HTTP_404_NOT_FOUND)
     else:
-        response = render(request, 'seed/404.html', {})
```
Comment (nllong) [inline seed/views/main.py:109]: can we delete the `404.html` file from the repo?

File: seed/views/main.py [hunk 3]
```diff
@@ -99,29 +99,23 @@ def version(request):
...
 
 def error500(request):
-    # Okay, this is a bit of a hack. Needed to move on.
     if '/api/' in request.path:
         return JsonResponse({
             "status": "error",
             "message": "Internal server error",
         }, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
     else:
-        response = render(request, 'seed/500.html', {})
```
Comment (nllong) [inline seed/views/main.py:122]: can we delete the `500.html` file from the repo?

**Review by nllong — APPROVED** [review]
looks good!  I didn't see a pop up when redirected, was I supposed to?

**Comment by perryr16** [comment]
> looks good! I didn't see a pop up when redirected, was I supposed to?
@nllong, yes, you should have seen a red warning like this:
Screen Shot 2022-03-02 at 11 43 06 AM

**Comment by nllong** [comment]
let me flush cache

**Comment by nllong** [comment]
> > looks good! I didn't see a pop up when redirected, was I supposed to?
> > @nllong, yes, you should have seen a red warning like this:
> 
> Screen Shot 2022-03-02 at 11 43 06 AM

hmm, I'm still not seeing it. `DEBUG=True`

**Comment by perryr16** [comment]
> hmm, I'm still not seeing it. `DEBUG=True`

Sorry, my testing notes had a mistake when testing on localhost,

/config/settings/docker_dev.py 
`DEBUG` needs to be set to `False`

/docker-compose.dev.yml 
add an `--insecure` flag
`web:`
... `command: python3 manage.py runserver --insecure --settings=config.settings.docker_dev 0.0.0.0:80`

**Comment by nllong** [comment]
I still can't see the popup, not sure what is going on. If @Ryoken can review this and confirm that it shows up, then we can merge.

**Review by Ryoken — APPROVED** [review]

**Comment by Ryoken** [comment]
> I still can't see the popup, not sure what is going on. If @Ryoken can review this and confirm that it shows up, then we can merge.

I was able to get it loading locally but it needed some cache clearing and docker rebooting... the usual nonsense!

**Pull request merged**